### PR TITLE
add-supertokens-to-authentication.md

### PR DESCRIPTION
**Pull request by kant01ne**
Hey team. This PR is a follow up on https://github.com/vercel/next.js/pull/21384 to add a reference to `with-supertokens` example to authentication doc.

### CODE REVIEW

**Comment by rishabhpoddar** [comment]
Is there anything we can do to improve our PR please? Thanks

File: docs/authentication.md [hunk 1]
```diff
@@ -130,6 +130,17 @@ Both of these libraries support either authentication pattern. If you're interes
 - [with-passport](https://github.com/vercel/next.js/tree/canary/examples/with-passport)
 - [with-passport-and-next-connect](https://github.com/vercel/next.js/tree/canary/examples/with-passport-and-next-connect)
 
+### SuperTokens
```
Comment (leerob) [inline docs/authentication.md:133]: Could you move this down to the bottom of the file please? I don't want to show preference by adding a new provider at the top.

File: docs/authentication.md [hunk 2]
```diff
@@ -130,6 +130,17 @@ Both of these libraries support either authentication pattern. If you're interes
...
+
+<details open>
+  <summary><b>Examples</b></summary>
+  <ul>
+    <li><a href="https://github.com/vercel/next.js/tree/canary/examples/with-supertokens">with-supertokens</a></li>
+  </ul>
+</details>
+
+[SuperTokens](https://supertokens.io) is an open source secure authentication system with many features such as credentials login, email verification, reset password flows, and third party logins.
```
Comment (leerob) [inline docs/authentication.md:142]: Maybe in your one-sentence summary talk about why you'd chose it over the others? Open source is probably important. "secure authentication system" is likely implied.

File: examples/with-supertokens/package.json [hunk 3]
```diff
@@ -11,8 +11,8 @@
     "next": "latest",
     "react": "17.0.1",
     "react-dom": "17.0.1",
-    "supertokens-auth-react": "^0.5.1",
-    "supertokens-node": "^3.4.0"
+    "supertokens-auth-react": "^0.6.0",
```
Comment (leerob) [inline examples/with-supertokens/package.json:14]: Thanks 🙏

File: docs/authentication.md (hunk 1 above)
Comment (NkxxkN) [reply docs/authentication.md]: Of course that makes sense! Done now :)

File: docs/authentication.md (hunk 2 above)
Comment (NkxxkN) [reply docs/authentication.md]: Good advice! I've updated to reflect that SuperTokens takes a modular approach while being customisable.

File: docs/authentication.md [hunk 4]
```diff
@@ -200,6 +189,18 @@ You can either use [FirebaseUI](https://github.com/firebase/firebaseui-web-react
...
+### SuperTokens
+
+<details open>
+  <summary><b>Examples</b></summary>
+  <ul>
+    <li><a href="https://github.com/vercel/next.js/tree/canary/examples/with-supertokens">with-supertokens</a></li>
+  </ul>
+</details>
+
+[SuperTokens](https://supertokens.io) is a highly customisable, open source auth split into modules so that you use only what you need.
```
Comment (leerob) [inline docs/authentication.md:201]:
```suggestion
[SuperTokens](https://supertokens.io) is a highly customizable, open-source solution split into modules (so you only use what you need).
```

File: docs/authentication.md [hunk 5]
```diff
@@ -200,6 +189,18 @@ You can either use [FirebaseUI](https://github.com/firebase/firebaseui-web-react
...
+SuperTokens currently supports credentials login, email verification, reset password flows and third party logins.
```
Comment (leerob) [inline docs/authentication.md:202]:
```suggestion
SuperTokens currently supports credentials login, email verification, password reset flows, and third-party logins.
```

**Review by leerob — APPROVED** [review]

**Comment by NkxxkN** [comment]
Thanks again for the review and update  🙏

**Pull request merged**